### ISSUE TITLE
postgres: Unparsable section: 'with csv header' in COPY statement

### ISSUE THREAD

**Issue by mtyszkiewicz** [sqlfluff/sqlfluff#4993]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

It looks like [CopyStatementSegment](https://github.com/sqlfluff/sqlfluff/blob/0730cd1e885e8942b62dab0aca2c2e0e94c72c11/src/sqlfluff/dialects/dialect_postgres.py#L4548) does not fully implement [Postgres spec](https://www.postgresql.org/docs/15/sql-copy.html).

### Expected Behaviour

copy copy_statement_bug from stdin with (format 'csv', header 'true');
copy copy_statement_bug from stdin with csv header;

both are valid examples of postgres syntax and both are rejected by sqlfluff

### Observed Behaviour

> sqlfluff lint test.sql
== [test.sql] FAIL                                                                                                                       
L:   6 | P:   1 |  PRS | Line 6, Position 1: Found unparsable section: 'copy                                                             
                       | copy_statement_bug from stdin with ...'
WARNING: Parsing errors found and dialect is set to 'ansi'. Have you configured your dialect correctly?
All Finished 📜 🎉!

## Running the copy data on the database
1. Create test.sql file:
```sql
create table if not exists copy_statement_bug (
    bug_id number,
    bug_report varchar(1000)
);
truncate table copy_statement_bug;
copy copy_statement_bug from stdin with csv header;
```
2. Create test.csv file:
```csv
bug_id,bug_report
1,aaaaaaaaaaaaaaaaaaaaaaaa
2,bbbbbbbbbbbbbbbbbbbbbbbb
```
3. Load data to database
```bash
POSTGRES_HOST="localhost"
connection_string="postgres://$POSTGRES_USER:$POSTGRES_PASSWORD@$POSTGRES_HOST:$POSTGRES_PORT/$POSTGRES_DB"

cat ./test.csv | psql $connection_string --command "$(cat ./test.sql)"
```
## Running the sqlfluff linter
```sqlfluff lint test.sql```

### Dialect

postgres

### Version

sqlfluff, version 2.1.3
Python 3.9.2

### Configuration
```toml
[sqlfluff]
dialect = postgres
max_line_length = 150
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by EkberHasanov** [comment]
Hi guys, as I understand no one assigned for this issue, I would like to try it if u don't mind

**Comment by sanchitram1** [comment]
+1 on this. I also found that there's no support for unions inside the query passed to `copy`. in my example, this would yield an unparseable error:

```sql
COPY
    (
        SELECT
            x,
            y,
        FROM t1
        WHERE t1.z = true
        UNION
        SELECT
            x,
            y
        FROM t2
    )
TO STDOUT WITH CSV HEADER;
```

```sh
$ `L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'COPY\n`
```

However, this might be because of the difference between COPY and \COPY in postgres, and i'm not entirely sure which one is being captured by [CopyStatementSegment](https://github.com/sqlfluff/sqlfluff/blob/0730cd1e885e8942b62dab0aca2c2e0e94c72c11/src/sqlfluff/dialects/dialect_postgres.py#L4548). if preferred, I can make a more detailed issue about this specific case. 

On a side note, i'd like to take on the original issue captured here, unless #5181 addresses it.

**Comment by alanmcruickshank** [comment]
@EkberHasanov & @sanchitram1 - I'm going to tidy up #5181 and get it merged. I'm going to set that PR up to close this issue on merge - but if you think there's additional syntax that's missing (for example the `UNION` case that you've highlighted). Do please start a new issue, and if you're feeling keen, then we'd love to have you as a contributor! My hope is that by getting the long outstanding PR finally merged, it gives you a cleaner slate to work from.
